DXRMapFixups/DeusEx/Classes/DXRFixupM06.uc

```diff
@@ -953,8 +958,22 @@ function AnyEntryMapFixes()
             m.bLocked = false;
             m.bHighlight = true;
         }
+
         HandleJohnSmithDeath();
         FixMaggieMoveSpeed();
+
+        if (
```

Comment:
still not entirely sure about this, Maggie can just return home after talking to Bob, which would mean less changes compared to vanilla

```suggestion
        if ( dxr.flags.settings.starting_map >= 75 || dxr.flagbase.GetBool('MaggieFled') )
```

Reply:
Actually, it seems that she's always gone in vanilla once you've been to the sword room. I verified this experimentally, and also there's an `OrdersTrigger` with the tag `LeavingSwordRoom` that tells Maggie to leave. 

So this PR actually brings us closer to vanilla. If we wanted to get even closer, we could have her leave as soon as the sword's found anywhere. But I like making her leave when you get the ROM encoding.

![image](https://github.com/user-attachments/assets/6bd3f1e0-6745-4398-88cb-d98d65e87839)

Reply:
well I guess it depends what you mean by "this change", there's the part where it's based on start_map (which should assume progress on main goals)

and the part where it's based on what maps you've visited (which doesn't always line up with goals the player has completed), like you can visit versalife before finding the DTS, you can even get the rom encoding before the DTS

Reply:
Personally, I've come around to thinking just entering a map shouldn't make her leave. The world should generally change when you complete a goal, I feel.